src/server/scripts/Pandaria/zone_the_wandering_isle.cpp

```diff
@@ -400,10 +402,34 @@ struct npc_instructor_zhi : public ScriptedAI
     TaskScheduler _scheduler;
 };
 
+enum EdictOfTemperanceSpells
+{
+    SPELL_FORCED_OPEN_EDICT_OF_TEMPERANCE = 122484
+};
+
+// 210986 - Edict of Temperance
+struct go_edict_of_temperance : public GameObjectAI
+{
+    go_edict_of_temperance(GameObject* go) : GameObjectAI(go) { }
+
+    bool OnReportUse(Player* /*player*/) override
+    {
+        me->SendGameObjectDespawn();
```

Comment:
use `me->DespawnOrUnsummon`, it also takes care of GameObjectDespawn packet. You might want to set a forced respawntime here btw

(same for spell hit)

Reply:
ok now works as intended